public/js/reducers/pause.js

```diff
@@ -59,6 +60,21 @@ function update(state = initialState, action, emit) {
     case constants.PAUSE_ON_EXCEPTIONS:
       const toggle = action.toggle;
       return state.set("shouldPauseOnExceptions", toggle);
+
+    case constants.ADD_EXPRESSION:
+      return state.setIn(["expressions", action.id],
+        { id: action.id,
+          expression: action.expression,
+          value: action.value || null });
+
+    case constants.EVALUATE_EXPRESSION:
+      if (action.status === "done") {
+        return state.setIn(["expressions", action.id],
```

Comment:
I'm not sure i f you want to use an array or a map. I suppose both work here, but it'd be bad if the array became sparse when expressions are removed...